src/Things/Structures/Structure.h

```diff
@@ -157,6 +157,7 @@ class Structure: public Thing
 
 	void setPopulationRequirements(const PopulationRequirements& pr) { mPopulationRequirements = pr; }
 
+	virtual int calculateIntegrity() = 0;
```

Comment:
I noticed a distinct lack of other direct pure `virtual` methods being declared in this class.

Given the large number of structures, and given that many of them will have very similar behaviour, I suspect much of the building integrity behaviour will be implemented in the base class, with just a few parameter values, where the parameter values can vary slightly between classes. That may be an easier way to control behaviour. It would certainly require less duplication of code in derived classes.

I don't oppose this method being `virtual`, but I suspect we don't want it to be pure virtual. I also don't think we need or want to override it in the vast majority of classes. There probably should be a good default implementation that comes from the base class.

Reply:
Removed Pure Virtual designation. Default behavior is to `return mIntegrity - calculateIntegrityDecayRate();` where `calculateIntegrityDecayRate` by default just returns `0`.